bot/cogs/Calculator.py

```diff
@@ -0,0 +1,153 @@
+import logging
+import re
+import discord
+import discord.ext.commands as commands
+
+log = logging.getLogger(__name__)
+
+
+class Calculator(commands.Cog):
```

Comment:
From what i can tell multiplication next to parens does not work, EG $calc 2(2+2) should be 8, but returns 4

Reply:
Yes it does not implicitly detect that type of multiplication. This algorithm requires there be a * for multiplication to occur.

Reply:
If the algorithim itself doesnt do it then we should preprocess the string to detect implicit multiplication and insert the * before we send it to the parse functions